addons/l10n_in_ewaybill_irn/models/account_move.py

```diff
@@ -0,0 +1,45 @@
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def action_l10n_in_ewaybill_create(self):
+        self.ensure_one()
+        action = super().action_l10n_in_ewaybill_create()
+        is_process_through_irn = (
+            self.move_type != 'out_refund'
+            and not self.debit_origin_id
+            and self.l10n_in_edi_status == 'sent'
+        )
+        action['context'] = {
+            'default_account_move_id': self.id,
+            'default_is_process_through_irn': is_process_through_irn,
+        }
```

Comment:
```suggestion
        action['context']['default_is_process_through_irn'] = is_process_through_irn
```
Why overwriting the context?